packages/xy-chart/src/series/LineSeries.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { curveCardinal, curveLinear } from '@vx/curve';
+import { GlyphDot } from '@vx/glyph';
+import { LinePath } from '@vx/shape';
+
+import { callOrValue } from '../utils/chartUtils';
+import { colors } from '../theme';
+import { lineSeriesDataShape } from '../utils/propShapes';
+
+const propTypes = {
+  data: lineSeriesDataShape.isRequired,
+  interpolation: PropTypes.oneOf(['linear', 'cardinal']), // @todo add more
+  label: PropTypes.string.isRequired,
+  showPoints: PropTypes.bool,
+
+  stroke: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
+  strokeDasharray: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
+  strokeWidth: PropTypes.oneOfType([PropTypes.func, PropTypes.number]),
+
+  // these will likely be injected by the parent chart
+  xScale: PropTypes.func.isRequired,
+  yScale: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  interpolation: 'cardinal',
+  showPoints: false,
+  stroke: colors.default,
+  strokeDasharray: null,
+  strokeWidth: 3,
+};
+
+export default function LineSeries({
+  data,
+  interpolation,
+  label,
+  showPoints,
+  stroke,
+  strokeDasharray,
+  strokeWidth,
+  xScale,
+  yScale,
+}) {
+  const x = d => d.x;
+  const y = d => d.y;
+  return (
+    <LinePath
+      key={label}
+      data={data}
+      xScale={xScale}
+      yScale={yScale}
+      x={x}
+      y={y}
+      stroke={callOrValue(stroke)}
+      strokeWidth={callOrValue(strokeWidth)}
+      strokeDasharray={callOrValue(strokeDasharray)}
+      curve={interpolation === 'linear' ? curveLinear : curveCardinal}
+      glyph={showPoints && ((d, i) => (
+        <GlyphDot
+          key={`${label}-${i}-${x(d)}`}
+          cx={xScale(x(d))}
+          cy={yScale(y(d))}
+          r={4}
+          fill={d.stroke || callOrValue(stroke, d, i)}
```

Comment:
the api will change for `callOrValue` it will move to `callOrValue(propName, data)` data should be a `{}`

but you don't have to worry about this for now since you've locked versions.